bot/src/discord/schedule_helper.js

```diff
@@ -0,0 +1,306 @@
+  if (rminutes > 0) res += `${rminutes}m`;
+  if (res !== '' && rseconds > 0) res += ' ';
+  if (rseconds > 0) res += `${rseconds}s`;
+  return res;
+}
+
+/**
+ * Gets the next time the command should run
+ * @param {number} start
+ * @param {number} frequency
+ */
+function getNextTime(start, frequency) {
+  const now = (new Date()).getTime();
+  if (now > start) {
+    return formatFrequency(frequency - ((now - start) % frequency));
+  }
+  return formatFrequency(start - now);
+}
+
+/**
+ * Creates or edits a schedule
+ * @param {string} suffix
+ * @param {Eris.Message} msg
+ */
+async function setSchedule(suffix, msg) {
+  const serverId = msg.guildID;
+  const channelId = msg.channel.id;
+
+  let serverSchedules = await WordScheduleModel.find({ serverId });
+
+  if (!serverSchedules) {
+    serverSchedules = [];
+  }
+
+  let freq = '';
+  let start = '';
+  const suffixArray = suffix.split(' ')
+    .map(s => s.trim().toLowerCase())
+    .filter(s => s);
+  const { length } = suffixArray;
+
+  if (length === 1) {
+    let index;
+    switch (suffixArray[0]) {
+      case 'clear':
+        await WordScheduleModel.deleteMany({ serverId });
+        return msg.channel.createMessage('All scheduled commands have been cleared.');
+
+      case 'list':
+        if (serverSchedules.length > 0) {
+          const list = serverSchedules
+            .map(i => `**Channel:** ${msg.channel.guild.channels.get(i.id).name}, **Frequency:** ${formatFrequency(i.frequency)}, **Start time:** ${new Date(i.start).toLocaleString()}, **Level:** ${i.level ? i.level : 'any'}, **Status:** ${i.status}, **Next word in** ${getNextTime(i.start, i.frequency)}.`)
+            .join('\n');
+          return msg.channel.createMessage(list);
+        }
+        return msg.channel.createMessage('List is empty.');
+
+      case 'stop':
+        index = serverSchedules.findIndex(i => i.id === channelId);
+        if (index !== -1) {
+          clearTimeout(serverSchedules[index].interval);
+          await WordScheduleModel.deleteOne({ _id: channelId });
+          return msg.channel.createMessage('Scheduled command stopped succesfully.');
+        }
+        return msg.channel.createMessage('Error: Scheduled command not found in this channel.');
+
+      case 'pause':
+        index = serverSchedules.findIndex(i => i.id === channelId);
+        if (index !== -1) {
+          if (serverSchedules[index].status === statusConstants.paused) {
+            return msg.channel.createMessage('Scheduled command is already paused.');
+          }
+          serverSchedules[index].status = statusConstants.paused;
+          await serverSchedules[index].save(); // put it into the database
+          return msg.channel.createMessage('Scheduled command paused succesfully.');
+        }
+        return msg.channel.createMessage('Error: Scheduled command not found in this channel.');
+
+      case 'pauseall':
+        await serverSchedules.forEach(async (schedule, i) => {
```

Comment:
This won't actually wait until they're all done, since forEach returns nothing. If you want to do that you can do

```js
await Promise.all(serverSchedules.map(async (schedule, i) => {
  ...
}));
```